internal/server/middleware.go

```diff
@@ -138,19 +149,40 @@ func unauthenticatedMiddleware(srv *Server) gin.HandlerFunc {
 	}
 }
 
+func orgRequired() gin.HandlerFunc {
+	return func(c *gin.Context) {
+		org := data.OrgFromContext(c.Request.Context())
+
+		if org == nil {
+			sendAPIError(c, internal.ErrBadRequest)
+		}
+	}
+}
+
+func setOrganizationInCtx(srv *Server) gin.HandlerFunc {
```

Comment:
if you want more info on this workflow, see https://mermaid.live/edit#pako:eNp1k01u2zAQha8y4KoFnAsYRQJachJvGqDpppC8YMiRRVgiVf4gCGTfvUPJsdk40YoYfHzz5g01MmkVsiVrOvsqW-EC_C5rA_TxamOk7bXZwS_8G9GHLdzc3AIvajMTPr7snBha4DG0aIKWImhrQBgF1u1Atij3M5q-y4nzcW3UYLUJ-V1Ud8ccmtod_qA_AF9VjdAd6GZSNjaAx7C9pn_aA6xGfFd35Fw79OlWLr7KtIuxtESc7_g4DN0bNGKPoEQQ8OPF3b6SSSCnDkF76j_ZmEj9v-3ioryuEkS4HVIuoltAS9l06GCHBh1N7LM2NFuvvafAt5-FVpwH5KurQdLYZfV0imYKHtVJhgabDx-2luhn7FAmc1mSxbjxX47HLzaAl4n0QvjPiHlvD1X0eFZKsvnSyhx9rL5YbZm1vJ_0FDYiduGj3P38QDdZ6fG69HBd2nyTlt6gifg9r84gP8fIFqxH1wut6I8ZU7lm9CZ6rNmSjkq4fc1qcyQuDrRTXCsdrGPLRnQeF0zEYJ_fjGTL4CK-Q6UWtJD-RB3_AcFeE6I